### PR TITLE
The holy Trinity!!! (command module)

### PR DESCRIPTION
**Pull request by J35P312**
I'm back!

### CODE REVIEW

File: lib/MIP/Program/Variantcalling/Trinity.pm [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+##          : $stdoutfile_path                 => Stdoutfile path
+
+    my ($arg_href) = @_;
+
+    ## Flatten argument(s)
+    my $FILEHANDLE;
+    my $infile_path;
+    my $max_intron_distance;
+    my $max_memory;
+    my $n_cpu;
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/Trinity.pm]: sacrifice brevity for readability: n_cpu -> number_core

File: lib/MIP/Program/Variantcalling/Trinity.pm [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            store       => \$minimum_number_supporting_pairs,
+            strict_type => 1,
+        },
+        max_memory => {
+            default     => 50,
+            store       => \$max_memory,
+            strict_type => 1,
+        },
+        n_cpu =>
+          { default => 16, store => \$outfile_path_prefix, strict_type => 1, },
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/Trinity.pm]: wrong store!

Always add a test template to your command module: https://github.com/Clinical-Genomics/MIP/blob/develop/templates/code/test_commands.t

File: lib/MIP/Program/Variantcalling/Trinity.pm [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        },
+        stdoutfile_path => {
+            store       => \$stdoutfile_path,
+            strict_type => 1,
+        },
+    };
+
+    check( $tmpl, $arg_href, 1 ) or croak q{Could not parse arguments!};
+
+    ## tiddit
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/Trinity.pm]: No trinity!

File: lib/MIP/Program/Variantcalling/Trinity.pm [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    my @commands = qw{ Trinity };
+
+    ## Infile
+    push @commands, q{--genome_guided_bam} . $SPACE . $infile_path;
+    ## cpu
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/Trinity.pm]: Add newline for between options. Whitespace makes it easier to read

**Comment by J35P312** [comment]
Done!

File: t/trinity_genome_guided.t [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+
+## Constants
+Readonly my $COMMA               => q{,};
+Readonly my $NEWLINE             => qq{\n};
+Readonly my $SPACE               => q{ };
+Readonly my $MAX_INTRON_DISTANCE => 10_000;
+Readonly my $CPU                 => 16;
+
+### User Options
+GetOptions(
```
Comment (henrikstranneheim) [inline t/trinity_genome_guided.t]: This is using an old template. Please use the new template instead. This part is imported as a cli and there are several improvement in imports and checks.

File: t/trinity_genome_guided.t [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+            exit_code => 1,
+        }
+    )
+  );
+
+BEGIN {
+
+### Check all internal dependency modules and imports
+## Modules with import
+    my %perl_module = ( q{MIP::Script::Utils} => [qw{ help }], );
```
Comment (henrikstranneheim) [inline t/trinity_genome_guided.t]: Chcek new template

File: t/trinity_genome_guided.t [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+use Modern::Perl qw{ 2014 };
+use Readonly;
+
+## MIPs lib/
+use lib catdir( dirname($Bin), q{lib} );
+use MIP::Script::Utils qw{ help };
+
+our $USAGE = build_usage( {} );
+
+my $VERBOSE = 1;
```
Comment (henrikstranneheim) [inline t/trinity_genome_guided.t]: This is using an old template. Please use the new template instead.

File: t/trinity_genome_guided.t [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+            function_base_commands_ref => \@function_base_commands,
+            module_function_cref       => $module_function_cref,
+            required_argument_href     => \%required_argument,
+        }
+    );
+}
+
+done_testing();
+
+######################
```
Comment (henrikstranneheim) [inline t/trinity_genome_guided.t]: Remove by using new template

**Comment by J35P312** [comment]
Tada!

**Comment by henrikstranneheim** [comment]
👍

**Pull request merged**